app/src/main/java/com/example/habitsmasher/ui/dashboard/EditHabitDialog.java

```diff
@@ -0,0 +1,133 @@
+package com.example.habitsmasher.ui.dashboard;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.example.habitsmasher.DaysTracker;
+import com.example.habitsmasher.DatePickerDialogFragment;
+import com.example.habitsmasher.Habit;
+import com.example.habitsmasher.HabitDialog;
+import com.example.habitsmasher.PublicPrivateButtons;
+import com.example.habitsmasher.R;
+
+import java.util.Date;
+
+/**
+ *  UI Class that represents and specifies the behaviour of the dialog
+ *  that is spawned when a user wants to edit a habit in their habit list
+ * @author Jason Kim
+ * @version 1.0
+ */
+public class EditHabitDialog extends HabitDialog {
```

Comment:
Changed name of EditHabitFragment to EditHabitDialog for consistency (for some reason GitHub thinks I deleted the old file and replace it with a new one)